traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go

```diff
@@ -956,8 +956,11 @@ func (dss *TODSSDeliveryService) Read(h http.Header, useIMS bool) ([]interface{}
 	for _, ds := range dses {
 		if version.Major > 3 && version.Minor >= 0 {
 			ds = ds.RemoveLD1AndLD2()
+			returnable = append(returnable, ds)
+		} else {
+			dsV3 := ds.DowngradeToV3()
+			returnable = append(returnable, dsV3)
```

Comment:
all API versions use this handler, so you need to downgrade to v3.0 and v2.0 as well when appropriate

Reply:
`DowngradeToV3` downgrades to an APIv3.1 structure - _even though it's named `DeliveryServiceNullableV30`!_

From that type's GoDoc:

> Again, this type is an alias that refers to the LATEST MINOR VERSION of API version 3 - NOT API version 3.0 as the name might imply.